File: Lab2.2/Circle.cpp
#include "Circle.h"
#include <cmath>
#include <sstream>

const double PI = 3.14159265358979323846;

Circle::Circle() : Figure("Circle"), centerX(0), centerY(0), radius(1) {
}

Circle::Circle(double centerX, double centerY, double radius)
    : Figure("Circle"), centerX(centerX), centerY(centerY) {
    // Ensure radius is positive
    this->radius = (radius > 0) ? radius : 1.0;
}

Circle::Circle(const Circle& other)
    : Figure(other), centerX(other.centerX), centerY(other.centerY), radius(other.radius) {
}

Circle::~Circle() {
}

double Circle::getRadius() const { return radius; }
double Circle::getCenterX() const { return centerX; }
double Circle::getCenterY() const { return centerY; }

double Circle::calculateArea() const {
    return PI * radius * radius;
}

double Circle::calculatePerimeter() const {
    return 2 * PI * radius;
}

std::string Circle::toString() const {
    std::stringstream ss;
    ss << "Circle: Center at (" << centerX << "," << centerY << "), Radius = " << radius;
    return ss.str();
}

File: Lab2.2cs/Circle.cs
using System;

namespace Lab2_2
{
    public class Circle : Figure
    {
        private double _radius;
        private double _centerX, _centerY;

        public double Radius { get { return _radius; } }
        public double CenterX { get { return _centerX; } }
        public double CenterY { get { return _centerY; } }

        public Circle() : base("Circle")
        {
            _centerX = 0;
            _centerY = 0;
            _radius = 1;
        }

        public Circle(double centerX, double centerY, double radius) : base("Circle")
        {
            _centerX = centerX;
            _centerY = centerY;
            _radius = radius > 0 ? radius : 1;
        }

        public Circle(Circle other) : base("Circle")
        {
            _centerX = other._centerX;
            _centerY = other._centerY;
            _radius = other._radius;
        }

        public override double CalculateArea()
        {
            return Math.PI * _radius * _radius;
        }

        public override double CalculatePerimeter()
        {
            return 2 * Math.PI * _radius;
        }

        public override string ToString()
        {
            return $"Circle: Center at ({_centerX},{_centerY}), Radius = {_radius}";
        }
    }
}
